lib/user.rb

```diff
@@ -0,0 +1,23 @@
+class User
+	attr_reader :name, :jokes
+	
+	def initialize(name)
+		@name = name
+		@jokes = []
+	end
+
+	def learn(joke)
+		@jokes << joke
+	end
+	
+	def joke_by_id(desired_id)
+		correct_joke = @jokes.select { |joke| joke.id == desired_id}
```

Comment:
`select` will always return an array, but the interaction pattern was asking for a specific instance of a joke as the return value, so this isn't quite meeting the requirements. `detect` or its alias `find` would be a better tool here. 

Nice single line syntax, and very close to the right tool for the job.